test/model/OauthCredentialsFactoryTest.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+namespace oat\taoOauth\test\model;
+
+use oat\generis\test\TestCase;
+use oat\taoOauth\model\provider\Provider;
+use oat\taoOauth\model\storage\grant\AuthorizationCodeType;
+use oat\taoOauth\model\storage\grant\ClientCredentialsType;
+use oat\taoOauth\model\storage\grant\PasswordType;
+use oat\taoOauth\model\storage\OauthCredentialsFactory;
+
+/**
+ * Class OauthCredentialsFactoryTest
+ */
+class OauthCredentialsFactoryTest extends TestCase
+{
+    /**
+     * @throws \common_exception_ValidationFailed
+     */
+    public function testGettingCredentialTypeByCredentials()
```

Comment:
Maybe this doesn't matter here but you should have a separate test method for each expected call from this method. You never know if for a newly created object this method will not behave differently. You also need to perform expect throw error assertion.

Reply:
got it. My bad